### PR TITLE
Create a service to discover resource link id

### PR DESCRIPTION
**Pull request by gabrielfs7**
Create a service to discover resource link id:

https://oat-sa.atlassian.net/browse/NEX-1834

### CODE REVIEW

File: model/Tool/Service/ResourceLinkIdDiscover.php [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+class ResourceLinkIdDiscover extends ConfigurableService implements ResourceLinkIdDiscoverInterface
+{
+    use OntologyAwareTrait;
+
+    public function discoverByDeliveryExecutionAndLtiProvider(
+        DeliveryExecution $execution,
+        LtiProvider $ltiProvider
+    ): string
+    {
+        if ($ltiProvider instanceof TenantLtiProvider) {
```
Comment (boajer) [inline model/Tool/Service/ResourceLinkIdDiscover.php]: We should avoid circular dependencies with `taoDeliverConnect`

**Review by boajer — CHANGES_REQUESTED** [review]

File: model/Tool/Service/ResourceLinkIdDiscover.php (hunk 1 above)
Comment (gabrielfs7) [reply model/Tool/Service/ResourceLinkIdDiscover.php]: Yes, you are right! I am checking a way to overcome this ;)

Comment (gabrielfs7) [reply model/Tool/Service/ResourceLinkIdDiscover.php]: Solved the issue. Thanks! https://github.com/oat-sa/extension-tao-deliver-connect/pull/79/files

File: model/Tool/Factory/LisOutcomeServiceUrlFactory.php [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+namespace oat\taoLtiConsumer\model\Tool\Factory;
+
+use oat\generis\model\OntologyAwareTrait;
+use oat\oatbox\service\ConfigurableService;
+use oat\tao\helpers\UrlHelper;
+
+class LisOutcomeServiceUrlFactory extends ConfigurableService
+{
+    use OntologyAwareTrait;
```
Comment (siwane) [inline model/Tool/Factory/LisOutcomeServiceUrlFactory.php]: Do u need this trait?

File: model/Tool/Service/ResourceLinkIdDiscover.php [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+namespace oat\taoLtiConsumer\model\Tool\Service;
+
+use oat\generis\model\OntologyAwareTrait;
+use oat\oatbox\service\ConfigurableService;
+use oat\taoDelivery\model\execution\DeliveryExecution;
+use oat\taoLtiConsumer\model\delivery\container\LtiDeliveryContainer;
+
+class ResourceLinkIdDiscover extends ConfigurableService implements ResourceLinkIdDiscoverInterface
+{
+    use OntologyAwareTrait;
```
Comment (siwane) [inline model/Tool/Service/ResourceLinkIdDiscover.php]: Do u need this trait?

File: test/unit/model/Tool/Factory/LisOutcomeServiceUrlFactoryTest.php [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program; if not, write to the Free Software
+ *  Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ *  Copyright (c) 2020 (original work) Open Assessment Technologies SA
+ */
+
```
Comment (siwane) [inline test/unit/model/Tool/Factory/LisOutcomeServiceUrlFactoryTest.php:19]: missing strict types declaration

File: test/unit/model/Tool/Service/ResourceLinkIdDiscoverTest.php [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program; if not, write to the Free Software
+ *  Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ *  Copyright (c) 2020 (original work) Open Assessment Technologies SA
+ */
+
```
Comment (siwane) [inline test/unit/model/Tool/Service/ResourceLinkIdDiscoverTest.php:19]: missing strict types declaration

**Review by siwane — COMMENTED** [review]
- [x] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [x] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful

File: model/Tool/Factory/LisOutcomeServiceUrlFactory.php (hunk 2 above)
Comment (gabrielfs7) [reply model/Tool/Factory/LisOutcomeServiceUrlFactory.php]: No, will fix. Thanks! :)

File: model/Tool/Service/ResourceLinkIdDiscover.php (hunk 3 above)
Comment (gabrielfs7) [reply model/Tool/Service/ResourceLinkIdDiscover.php]: No, will fix. Thanks! :)

**Review by siwane — APPROVED** [review]
- [x] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [x] New code is not subject to concurrency issues (if applicable)
- [x] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful

**Pull request merged**